docs/guides/service_principal_configuration.md

```diff
@@ -18,55 +18,72 @@ We recommend using either a Service Principal or Managed Identity when running T
 
+---
+
 ## Azure Active Directory permissions
 
 Now that you have created and authenticated an Application / Service Principal pair, you will need to grant some permissions to administer Azure Active Directory. You can choose either of the following methods to achieve similar results.
 
-### Method 1: Directory Roles (recommended)
+### Method 1: API roles (recommended for service principals)
 
-With this method, you will assign directory roles to the Service Principal you created, to grant the desired permissions to administer objects in your Azure Active Directory tenant.
+This method involves granting API roles to your Application, and then granting consent for your Service Principal to access the APIs in its own capacity (i.e. not on behalf of a user).
 
-Navigate to the [**Azure Active Directory** overview][azure-portal-aad-overview] within the [Azure Portal][azure-portal]. Go to the [**Roles and Administrators** blade][azure-portal-aad-roles-blade].
+Navigate to the [Azure Active Directory overview][azure-portal-aad-overview] within the [Azure Portal][azure-portal] and select the [App Registrations blade][azure-portal-aad-applications-blade]. Locate your registered Application and click on its display name to manage it.
 
-Locate the role you wish to assign and click on it. Consult the [documentation for administrator role permissions][admin-roles-docs] from Microsoft for more information about the available roles and the permissions they grant.
+Go to the API Permissions blade for the Application and click the "Add a permission" button. In the pane that opens, select "Microsoft Graph".
 
-Click "Add assignments" and type the name of your Service Principal in the search box to locate it. If you know the Object ID of the Service Principal, verify that it is the same. Select it and click the "Add" button to assign the role.
+Choose "Application Permissions" for the permission type, and check the permissions you would like to assign. The permissions you need will depend on which directory objects you wish to manage with Terraform. The following table show the required permissions for some common resources:
 
-The choice of which directory roles to assign will be specific to your organisations Commonly used roles include:
+Resource(s) | Role Name(s)
+-------- | ---------------
+`data.azuread_application`<br>`data.azuread_service_principal` | Application.Read.All
+`data.azuread_domains` | Domain.Read.All
+`data.azuread_group`<br>`data.azuread_groups` | Group.Read.All
+`data.azuread_user`<br>`data.azuread_users` | User.Read.All
+`azuread_application`<br>`azuread_application_app_role`<br>`azuread_application_certificate`<br>`azuread_application_oauth2_permission_scope`<br>`azuread_application_password`<br>`azuread_service_principal`<br>`azuread_service_principal_certificate`<br>`azuread_service_principal_password` | Application.ReadWrite.All
+`azuread_group`<br>`azuread_group_member` | Group.ReadWrite.All
+`azuread_user` | User.ReadWrite.All
```

Comment:
Since the number of resources will increase as time goes on - this may make more sense on each data sources/resources page? As a Data Source will require Read but a Resource will require .All?

Reply:
100% agreed, I changed the wording above this table to indicate this is a subset of resources (I thought it would be useful to indicate the current set or perms on the migration guide), and intend to add a section to each resource/DS page with the corresponding permissions